R/ResamplePrediction.R

```diff
@@ -22,8 +22,8 @@ makeResamplePrediction = function(instance, preds.test, preds.train, task.desc)
   if (any(trnull)) pr.tr = preds.train[!trnull] else pr.tr = preds.train
 
   data = setDF(rbind(
-    rbindlist(lapply(seq_along(pr.te), function(X) cbind(pr.te[[X]]$data, iter = X, set = "test"))),
-    rbindlist(lapply(seq_along(pr.tr), function(X) cbind(pr.tr[[X]]$data, iter = X, set = "train")))
+    rbindlist(lapply(seq_along(pr.te), function(X) cbind(pr.te[[X]]$data, iter = X, set = "test")), fill = TRUE),
```

Comment:
you may want to consider using dplyr::bind_rows instead of rbindlist. It's suppose to replace it if I'm not mistaken

Reply:
We do not use _dplyr_ in _mlr_ but rather _data.table_. Both should do the same.

Reply:
This is one of the main problems. @HerrMo, do you remember why this was necessary?

Reply:
Taking the `tests/testthat/test_surv_kaplanmeier.R` test, would I arrive at this point when debugging?

Reply:
In short, this was implemented to be able to merge the results of successful and unsuccessful CV iterations (i.e., failure models and successful model fits). 
Some details:
Since in `rbindlist(lapply(seq_along(pr.te), function(X) cbind(pr.te[[X]]$data, iter = X, set = "test")))` the data.frames of the single iterations get merged to one single data.frame/data.table rowwise, the data.frames need to have the same number of columns. The problem with survival models is that unsuccessful iterations yield less columns than successful ones. Setting `fill = TRUE` solves this issue, but I'm afraid it's more of a workaround than a satisfactory solution.

Reply:
I have the fear that changing this globally could possibly trigger some unwanted side effects even if all tests will pass.

Would it be possible to add the missing columns to survival models manually? This way we would avoid the global change here and encapsulate possible consequences.

Reply:
Probably, but I have no idea how that works and if it's easy. 
I don't know exactly what the cause is, but I think the problem of the different number of columns shouldn't occur in the first place.